tests/framework/extensions/supportmatrix/versionchecks.go

```diff
@@ -0,0 +1,47 @@
+package supportmatrix
+
+import (
+	"strings"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	PodResourceSteveType = "pod"
+)
+
+// CheckSupportMatrixImageVersions checks the image versions needed for support matrix tests.
+func CheckSupportMatrixImageVersions(client *rancher.Client, clusterID string) error {
```

Comment:
Our main intent here is to just check the versions for the particular images, we aren't looking at functionality.

Reply:
I agree the `imagResults` is not even used. Would make sense to make these test validations within provisioning rather than being extensions? My point is we need to have automation validations, if that makes sense

Reply:
In that case we should return the imageResults since you're already getting them. What is the purpose of that variable?

Reply:
So again for this one as well, sorry, I was not talking about integration-level functionality. I was talking about this helper's functionality. Similar to the above comment, my point was, this needs a check step for those particular images

Reply:
I can remove the `imageResults`, that's honestly an oversight on my end from other similar code we have in the framework. With respect to the versions, essentially my concern is the same as in the etcd comment above.

I think we're all in agreement here about having validation checks here, but the current QA process is more so just checking the versioning of images as opposed to expecting something in particular.

Reply:
Yep, I can do that 😄

Reply:
Resolved in latest commit.